tls/s2n_kem.h

```diff
@@ -74,21 +74,32 @@ struct s2n_kem_group_params {
 };
 
 extern const struct s2n_kem s2n_kyber_512_r3;
+extern const struct s2n_kem s2n_kyber_768_r3;
+extern const struct s2n_kem s2n_kyber_1024_r3;
 
 /* x25519 based tls13_kem_groups require EVP_APIS_SUPPORTED */
-#if EVP_APIS_SUPPORTED
+/* Kyber758+ requires S2N_LIBCRYPTO_SUPPORTS_KYBER */
+#if defined(S2N_LIBCRYPTO_SUPPORTS_KYBER) && EVP_APIS_SUPPORTED
+    #define S2N_SUPPORTED_KEM_GROUPS_COUNT 6
+#elif defined(S2N_LIBCRYPTO_SUPPORTS_KYBER) && !EVP_APIS_SUPPORTED
+    #define S2N_SUPPORTED_KEM_GROUPS_COUNT 4
+#elif !defined(S2N_LIBCRYPTO_SUPPORTS_KYBER) && EVP_APIS_SUPPORTED
     #define S2N_SUPPORTED_KEM_GROUPS_COUNT 2
 #else
     #define S2N_SUPPORTED_KEM_GROUPS_COUNT 1
 #endif
```

Comment:
This has gotten REALLY complex. Can you do this in a cleaner way?

Reply:
hm, i'll see what I can do.

Reply:
this will be addressed in PR #4101